src/orion/core/worker/producer.py

```diff
@@ -95,12 +95,12 @@ def produce(self):
             self._sample_guard(start)
 
             log.debug("### Algorithm suggests new points.")
-            new_points = self.naive_algorithm.suggest(self.pool_size)
+            new_points = self.naive_algorithm.suggest()
```

Comment:
should we limit the number of points been suggested or been persisted smaller than `max_trials`, or there will be bunch of `new` trials in db but never executed?

Reply:
I would be in favor of registering all returned trials. We never know if some running trials will fail, forcing the producer to generate new trials. At least they would already be present. Does that seem reasonable?

Reply:
I was more wondering how this will be exposed to user. If there are too much `new` or `pend` status trials when they query the experiment status, may be a little bit confusing. Although I saw algorithms like `gridsearch` will actually limit the grid size to `max_trials`.

Reply:
Hum, but the algorithm should be able to sample more than max_trials if many of them fails. That's a problem with grid search by construction, it samples the grid and that's it. But for the other algorithms they can sample more. Maybe it should be something like `max(max_trials - self.n_suggested, 1)`? So that the algorithm can still return more than `max_trials` but overshoot as little as possible.

Reply:
yeah, this seems better.

Reply:
I opted for giving this responsibility to the producer since it knows how many broken trials there are and can adjust the request accordingly. The method suggest() would have no default, and algorithms would be expected to return as many points as possible, up to `num`. I would be happy to have your feedback on this.

Reply:
looks good to me. do you think it is necessary to validate the number of points suggested by the algorithm is actually smaller than request?

Reply:
I would rely on the unittests of the algorithms for this. If it ever happens that the algo returns more points it will be better if the producer registers them anyway, otherwise there will be issues with the algorithm thinking the additional trials were registered but they weren't.